tlscanary/js/scan_worker.js

```diff
@@ -186,6 +190,438 @@ function collect_request_info(xhr, report_certs) {
+        "805A2FD8": "SSL_ERROR_RX_MALFORMED_APPLICATION_DATA",
+        "805A2FD9": "SSL_ERROR_RX_MALFORMED_HANDSHAKE",
+        "805A2FDA": "SSL_ERROR_RX_MALFORMED_ALERT",
+        "805A2FDB": "SSL_ERROR_RX_MALFORMED_CHANGE_CIPHER",
+        "805A2FDC": "SSL_ERROR_RX_MALFORMED_FINISHED",
+        "805A2FDD": "SSL_ERROR_RX_MALFORMED_CLIENT_KEY_EXCH",
+        "805A2FDE": "SSL_ERROR_RX_MALFORMED_CERT_VERIFY",
+        "805A2FDF": "SSL_ERROR_RX_MALFORMED_HELLO_DONE",
+        "805A2FE0": "SSL_ERROR_RX_MALFORMED_CERT_REQUEST",
+        "805A2FE1": "SSL_ERROR_RX_MALFORMED_SERVER_KEY_EXCH",
+        "805A2FE2": "SSL_ERROR_RX_MALFORMED_CERTIFICATE",
+        "805A2FE3": "SSL_ERROR_RX_MALFORMED_SERVER_HELLO",
+        "805A2FE4": "SSL_ERROR_RX_MALFORMED_CLIENT_HELLO",
+        "805A2FE5": "SSL_ERROR_RX_MALFORMED_HELLO_REQUEST",
+        "805A2FE6": "SSL_ERROR_TX_RECORD_TOO_LONG",
+        "805A2FE7": "SSL_ERROR_RX_RECORD_TOO_LONG",
+        "805A2FE8": "SSL_ERROR_BAD_BLOCK_PADDING",
+        "805A2FE9": "SSL_ERROR_NO_CIPHERS_SUPPORTED",
+        "805A2FEA": "SSL_ERROR_UNKNOWN_CIPHER_SUITE",
+        "805A2FEB": "SSL_ERROR_FORTEZZA_PQG",
+        "805A2FEC": "SSL_ERROR_SSL_DISABLED",
+        "805A2FED": "SSL_ERROR_EXPIRED_CERT_ALERT",
+        "805A2FEE": "SSL_ERROR_REVOKED_CERT_ALERT",
+        "805A2FEF": "SSL_ERROR_BAD_CERT_ALERT",
+        "805A2FF0": "SSL_ERROR_BAD_MAC_ALERT",
+        "805A2FF1": "SSL_ERROR_BAD_MAC_READ",
+        "805A2FF2": "SSL_ERROR_SSL2_DISABLED",
+        "805A2FF3": "SSL_ERROR_POST_WARNING",
+        "805A2FF4": "SSL_ERROR_BAD_CERT_DOMAIN",
+        "805A2FF5": "SSL_ERROR_WRONG_CERTIFICATE",
+        "805A2FF7": "SSL_ERROR_UNSUPPORTED_VERSION",
+        "805A2FF8": "SSL_ERROR_UNSUPPORTED_CERTIFICATE_TYPE",
+        "805A2FF9": "SSL_ERROR_BAD_SERVER",
+        "805A2FFA": "SSL_ERROR_BAD_CLIENT",
+        "805A2FFC": "SSL_ERROR_BAD_CERTIFICATE",
+        "805A2FFD": "SSL_ERROR_NO_CERTIFICATE",
+        "805A2FFE": "SSL_ERROR_NO_CYPHER_OVERLAP",
+        "805A2FFF": "SSL_ERROR_US_ONLY_SERVER",
+        "805A3000": "SSL_ERROR_EXPORT_ONLY_SERVER",
+        "805A3FE7": "MOZILLA_PKIX_ERROR_CA_CERT_USED_AS_END_ENTITY",
+        "805A3FE8": "MOZILLA_PKIX_ERROR_INADEQUATE_KEY_SIZE",
+        "805A3FE9": "MOZILLA_PKIX_ERROR_V1_CERT_USED_AS_CA",
+        "805A3FEB": "MOZILLA_PKIX_ERROR_NOT_YET_VALID_CERTIFICATE",
+        "805A3FEC": "MOZILLA_PKIX_ERROR_NOT_YET_VALID_ISSUER_CERTIFICATE",
+        "805A3FED": "MOZILLA_PKIX_ERROR_SIGNATURE_ALGORITHM_MISMATCH",
+        "805A3FEE": "MOZILLA_PKIX_ERROR_OCSP_RESPONSE_FOR_CERT_MISSING",
+        "805A3FEF": "MOZILLA_PKIX_ERROR_VALIDITY_TOO_LONG",
+        "805A3FF0": "MOZILLA_PKIX_ERROR_REQUIRED_TLS_FEATURE_MISSING",
+        "805A3FF1": "MOZILLA_PKIX_ERROR_INVALID_INTEGER_ENCODING",
+        "805A3FF2": "MOZILLA_PKIX_ERROR_EMPTY_ISSUER_NAME",
+        "805A3FF3": "MOZILLA_PKIX_ERROR_ADDITIONAL_POLICY_CONSTRAINT_FAILED",
+        "805A3FFD": "MOZILLA_PKIX_ERROR_V1_CERT_USED_AS_CA"
```

Comment:
This is missing the newest error code `MOZILLA_PKIX_ERROR_SELF_SIGNED_CERT` and I'm adding ` MOZILLA_PKIX_ERROR_SELF_SIGNED_CERT` shortly. While I'd hope this list doesn't change a lot it does from time to time. So I'm wondering if there isn't a better way to do this.
You can get the TLS error code string with `securityInfo.errorCodeString` [nsITransportSecurityInfo](https://searchfox.org/mozilla-central/source/netwerk/socket/nsITransportSecurityInfo.idl#16).

Reply:
I don't have a good way for generic errors. Only for the TLS errors we hand out. Maybe it would make sense to handle the errors separately? The high level error Firefox gives out through transportSecurityInfo, which you can get as string, and the low level error, which uses a list? (Not sure if that makes sense, I don't know enough about canary :upside_down_face:) But what I would do is, in addition to writing unknown error when the error is not in the list, adding the error code for people to look up the error code if it is not in the list.